src/evm-block-extractor/src/main.rs

```diff
@@ -157,7 +215,46 @@ fn write_blocks(
     Ok(())
 }
 
+/// Scans the output dir for existing block files and returns the last block number
+fn get_last_block_number_from_output_dir(output_dir: &Path) -> Option<u64> {
+    // scan files in output dir
+    let mut last_block: Option<u64> = None;
+    let mut output_dir = std::fs::read_dir(output_dir).ok()?;
+    for file in output_dir.by_ref() {
+        let file = file.ok()?;
+        log::info!("found file: {}", file.path().display());
+        last_block = match (
+            last_block,
+            get_last_block_number_from_output_file(&file.path()),
+        ) {
+            (last_block, None) => last_block,
+            (Some(last_block), Some(file_last_block)) => Some(last_block.min(file_last_block)),
+            (None, Some(file_last_block)) => Some(file_last_block),
+        };
+    }
+
+    last_block
+}
+
+/// Returns the last block number from the given output file
 fn get_last_block_number_from_output_file(output_file: &Path) -> Option<u64> {
     let mut reader = BlocksReader::new(Path::new(output_file)).ok()?;
     reader.get_last_block_number().ok()
 }
+
+/// Returns the output file path for the given output dir based on the current time
+fn get_output_file(output_dir: &Path) -> PathBuf {
```

Comment:
we don't need a new file for each loop. One file a day or one file every hour is ok. Append to the file if it exists